modules/navigation/register_types.cpp

```diff
@@ -59,24 +70,38 @@ void initialize_navigation_module(ModuleInitializationLevel p_level) {
 		_nav_mesh_generator = memnew(NavigationMeshGenerator);
 		GDREGISTER_CLASS(NavigationMeshGenerator);
 		Engine::get_singleton()->add_singleton(Engine::Singleton("NavigationMeshGenerator", NavigationMeshGenerator::get_singleton()));
-#endif
+
+		GDREGISTER_CLASS(NavigationGeometryParser3D);
+		// add default 3D node navigation geometry parsers
+#ifdef MODULE_GRIDMAP_ENABLED
+		NavigationMeshGenerator::get_singleton()->register_geometry_parser_3d(memnew(GridMap3DNavigationGeometryParser3D));
```

Comment:
The registrations of the parsers should also ideally happen in the modules that they are needed for, to avoid every module needing an `ifdef` here.

Reply:
agree